clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -245,7 +248,8 @@ protected JoinGroupRequestData.JoinGroupRequestProtocolCollection metadata() {
             Subscription subscription = new Subscription(topics,
                                                          assignor.subscriptionUserData(joinedSubscription),
                                                          subscriptions.assignedPartitionsList(),
-                                                         generation().generationId);
+                                                         generation().generationId,
+                                                         rackId);
```

Comment:
Is it possible to add a unit test which verifies that `rackId` is correctly sent out in the `JoinGroup` request? I am not sure how feasible is this.

Reply:
Added a unit test. Also added an integration test for verifying propagation since we haven't yet added a rack-aware assignor.